Component/MvuDrawingApp/App.fs

```diff
@@ -0,0 +1,242 @@
+        { IsPressed: bool
+          LastPoint: Point option
+          Color: Color
+          Size: float }
+
+    type Msg =
+        | PointerPressed of Input.PointerPressedEventArgs
+        | PointerReleased of Input.PointerReleasedEventArgs
+        | PointerMoved of Input.PointerEventArgs
+
+    let init () =
+        { IsPressed = false
+          LastPoint = None
+          Color = Colors.Black
+          Size = 2. }
+
+    let canvasRef = ViewRef<Canvas>()
+
+    let update msg model =
+        match msg with
+        | PointerPressed _ -> { model with IsPressed = true }
+        | PointerReleased _ -> { model with IsPressed = false }
+        | PointerMoved args ->
+            let point = args.GetPosition(canvasRef.Value)
+
+            if model.IsPressed then
+                match model.LastPoint with
+                | Some lastPoint ->
+                    let brush = unbox(ColorToBrushConverter.Convert(box(model.Color), typeof<IBrush>))
+
+                    let line =
+                        Shapes.Line(StartPoint = lastPoint, EndPoint = point, Stroke = brush, StrokeThickness = model.Size, StrokeLineCap = PenLineCap.Round)
+
+                    if canvasRef.Value <> null then
+                        canvasRef.Value.Children.Add(line)
+
+                    { model with LastPoint = Some point }
+                | None -> { model with LastPoint = Some point }
+            else
+                { model with LastPoint = Some point }
+
+    let view () =
+        Canvas(canvasRef)
+            .verticalAlignment(VerticalAlignment.Stretch)
+            .horizontalAlignment(HorizontalAlignment.Stretch)
+            .background(SolidColorBrush(Colors.White))
+            .onPointerPressed(PointerPressed)
+            .onPointerReleased(PointerReleased)
+            .onPointerMoved(PointerMoved)
+
+
+module App =
+    let theme = FluentTheme()
+
+    type Model =
+        { Setting: Setting.Model
+          DrawingCanvas: DrawingCanvas.Model }
+
+    type Msg =
+        | SettingMsg of Setting.Msg
+        | DrawingCanvasMsg of DrawingCanvas.Msg
+
+    let init () =
+        { Setting = Setting.initModel
+          DrawingCanvas = DrawingCanvas.init() }
+
+    let update msg model =
+        match msg with
+        | SettingMsg msg ->
+            let setting = Setting.update msg model.Setting
+            { model with Setting = setting }
+
+        | DrawingCanvasMsg msg ->
+            let drawingCanvas =
+                { model.DrawingCanvas with
+                    Color = model.Setting.ColorPicker.Color
+                    Size = model.Setting.SizePicker.Size }
+
+            let drawingCanvas = DrawingCanvas.update msg drawingCanvas
+
+            { model with
+                DrawingCanvas = drawingCanvas }
+
+    let content (model: Model) =
+        (Dock() {
+            View.map SettingMsg (Setting.view(model.Setting).dock(Dock.Bottom))
+            View.map DrawingCanvasMsg (DrawingCanvas.view().dock(Dock.Top))
```

Comment:
@TimLariviere Im converting this sample the use the `Component` api. Any thoughts on how I can remove the `View.map` here and still be able get the updated data from each view ?.

Reply:
Not sure what you mean by the "updated data"?

If you make Setting and DrawingCanvas as components, it should allow you to remove `View.map` as Component returns a `WidgetBuilder<'msg, 'marker>` where `'marker` is the marker type of the root of the component and `'msg` is actually undefined. This undefined `'msg` will be inferred by F# at the call site to conform to the `'msg` type where it is being used.

```fs
// Returns a WidgetBuilder<'msg, IFabVStack>
// Note that here 'msg is an undefined generic type,
// it is not the same than the internal msg type of the component in case of MVU component
let myComponent() =
    Component() {
        VStack() {
            // ...
        }
    }

type MyWonderfulMsg = | Clicked

let program = ...

Component(program) {
    let! model = Mvu.State

    VStack() {
        Button("Click me", Clicked)
 
        // F# will infer that WidgetBuilder<'msg, IFabVStack>
        // needs to be WidgetBuilder<MyWonderfulMsg, IFabVStack>
        // to be able to compile here, since there is nothing preventing it,
        // that's exactly what F# will automatically do
        // Hence no need for View.map
        myComponent()
}
```

Reply:
Just like `Label("Hello")` will return a `WidgetBuilder<'msg, IFabLabel>` actually. Label by itself does not know what Msg to use, so it leaves F# decides automatically.